auth/basic/db.go

```diff
@@ -115,3 +141,39 @@ func (b *Store) Validate(ctx context.Context, username, password string) (string
 
 	return userID, nil
 }
+
+// ValidatePassword should check if the password matches the user's stored password
+func (b *Store) ValidatePassword(ctx context.Context, userID string, password string) (ValidatedPassword, error) {
+	err := validate.Many(
+		validate.UUID("UserID", userID),
+		validate.Text("oldPassword", password, 8, 200),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	row := b.getByID.QueryRowContext(ctx, userID)
+	var hash string
+	err = row.Scan(&hash)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, errors.New("invalid userID")
+		}
```

Comment:
nit: flatten also use `unknown` instead of `invalid` (we already ensured it's valid)
```suggestion
	if errors.Is(err, sql.ErrNoRows) {
		return nil, errors.New("unknown userID")
	}
	if err != nil {
```